analyzer/codechecker_analyzer/cmd/analyze.py

```diff
@@ -655,7 +675,47 @@ def __get_skip_handler(args):
 
     if skip_file_content:
         LOG.debug_analyzer("Creating skiplist handler.")
-        return skiplist_handler.SkipListHandler(skip_file_content)
+        handler = skiplist_handler.SkipListHandler(skip_file_content)
+
+        analyze_headers = []
+
+        # Check whether the skip file contains a header file which is not
+        # skipped.
+        for skip_line in handler.skip_file_lines:
+            if skip_line[0] == '+' and \
+              skip_line.lower().endswith((".h", ".hh", ".hpp")):
+
+                norm_skip_path = os.path.normpath(skip_line[1:].strip())
+                rexpr = re.compile(
+                    fnmatch.translate(norm_skip_path + '*'))
+                analyze_headers.append((skip_line, rexpr))
+
+        # Get source files which depend on the previously collected header
+        # files and create a new skip list handler where we include these
+        # files in the beginning.
+        if analyze_headers:
+            dependencies = __get_source_dependencies(compile_commands)
```

Comment:
I think an info message would nice here so the user will know what is happening. It takes a few seconds without any output message to collect the dependencies.